### DISCUSSION TITLE
tt.gz file conversion?

### DISCUSSION THREAD

**Discussion by ucigrangers** [dipy/dipy#2654] [1. Ideas & Feedback]
Hello. 

I am wondering if DIPY has any type of flexible conversion of at least the ability to read .tt.gz files? 

I see in the examples there are at least 5 file types that DIPY works with (I have used trk.gz files) but wondering if DIPY would consider adding the use of .tt.gz files? If not, does anyone know how to convert .tt.gz files to a file type that DIPY will accept? 

Thanks so much in advance!

**Comment by arokem** [comment]
Do you have any documentation about this file format? Which software produces it?

**Reply by ucigrangers** [reply]
DSI studio produces tt (tiny track) files as of the last few years as a default tractography file. They also produce trk files but not for all functions. There is information here regarding tt.gz files here: https://dsi-studio.labsolver.org/doc/gui_t3_atk.html 
and here: https://dsi-studio.labsolver.org/doc/cli_data.html

They look to be read more easily with matlab but I am hoping to read them into dipy.

**Reply by pjsjongsung** [reply]
A simple workaround  could be this. Extract it to a .tt file, rename its name to .mat and use the `scipy.io.loadmat` function.

**Reply by ucigrangers** [reply]
Thanks @pjsjongsung ! So...a little more info here. I am trying to use an example from dipy (https://dipy.org/documentation/1.5.0/examples_built/viz_bundles/#example-viz-bundles). Does the mat file have a similar property as accessing the "streamlines" array as if loading a trk file with nib.streamlines.load?

**Reply by skoudoro** [reply]
I think, this is an issue for nibabel: https://github.com/nipy/nibabel

we use nibabel to read our data and it will be better to add a request to nibabel. 

But thank you for this information, we will definitively look into it

**Reply by pjsjongsung** [reply]
I agree with @skoudoro, but if you need a fast solution, I would suggest loading both of them to compare their structure and looking at the code in the 'load/save tt files in matlab' section in the link you provided above.
